expo/tests/scheduler/test_edit_event.py

```diff
@@ -1,97 +1,329 @@
+                      str(room),
+                      response.content)
+        # check volunteers
+        self.assertIn('<input id="id_event-max_volunteer" min="0" ' +
+                      'name="event-max_volunteer" type="number" value="3" />',
+                      response.content)
+        # check description
+        self.assertIn('New Description', response.content)
+
+    def test_no_login_gives_error(self):
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        response = self.client.get(url, follow=True)
+        redirect_url = reverse('login', urlconf='gbe.urls') + "/?next=" + url
+        self.assertRedirects(response, redirect_url)
+        self.assertTrue(is_login_page(response))
+
+    def test_bad_user(self):
+        login_as(ProfileFactory(), self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 403)
+
+    def test_good_user_get_bad_event(self):
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk+1])
+        response = self.client.get(url, follow=True)
+        self.assertEqual(response.status_code, 404)
+
+    def test_good_user_get_success(self):
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        response = self.client.get(url)
+        assert_good_sched_event_form(response, self.context.bid)
+
+    def test_good_user_minimal_post(self):
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        form_data = get_sched_event_form(self.context)
+        response = self.client.post(
+            url,
+            data=form_data,
+            follow=True)
+        self.assert_good_post(response,
+                              self.context,
+                              self.context.days[0],
+                              self.context.room)
+
+    def test_good_user_invalid_submit(self):
+        login_as(self.privileged_profile, self)
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["Class", self.context.sched_event.pk])
+        form_data = get_sched_event_form(self.context)
+        form_data['event-location'] = 'bad room'
+        response = self.client.post(
+            url,
+            data=form_data)
 
         self.assertEqual(response.status_code, 200)
```

Comment:
Hm. This looks a lot like the assert_good_post code, except the values are just a little different.